src/Tools/dotnet-monitor/Auth/IAuthHandler.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Swashbuckle.AspNetCore.SwaggerGen;
+using Swashbuckle.AspNetCore.SwaggerUI;
+using System;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Auth
+{
+    internal interface IAuthHandler
+    {
+        void ConfigureApiAuth(IServiceCollection services, HostBuilderContext context);
+        void ConfigureSwaggerGenAuth(SwaggerGenOptions options);
+        void ConfigureSwaggerUI(SwaggerUIOptions options);
+        void LogStartup(ILogger logger, IServiceProvider serviceProvider);
```

Comment:
All of the other methods adjust configuration whereas this logs information post-configuration. I would be cleaner if this were either (a) folded into the ConfigureApiAuth where a IStartupLogger implementation is provided, or (b) refactored to `IStartupLogger CreateStartupLogger(ILogger, IServiceProvider);` so that the IStartupLogger instance can be inserted into the correct place in the `ConfigureStartupLoggers` method. This then allows `IAuthHandler` to not be registered as a service since it no longer executes any implementation itself for post-configuration executable and can be purely an implementation detail of configuration.